native/src/main/java/io/ballerina/stdlib/kafka/impl/KafkaRecordConsumer.java

```diff
@@ -177,4 +177,21 @@ public void immediateStopConsume() {
         }
         this.kafkaConsumer.close(Duration.ofMillis(0));
     }
+
+    /**
+     * Stops Kafka consumer polling cycles and forcefully shutdowns scheduled thread pool.
+     */
+    public void stopScheduledPollTask() {
+        closed.set(true);
+        this.kafkaConsumer.wakeup();
+        this.pollTaskFuture.cancel(true);
+        this.executorService.shutdownNow();
+        try {
+            this.executorService.awaitTermination(stopTimeout, TimeUnit.MILLISECONDS);
```

Comment:
Lets make 30 seconds.

Reply:
Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/310/commits/36c9dd547c14677dee1c4c474f7bc28846d084f1